### PR TITLE
Issue #2247 - Remove user-contact connection

### LINKED ISSUE #2247: Remove User-contact connection
### Issue description

We have a very interconnected architecture between contacts and users that even uses signals. User creates a contact and populates it with data, but then a contact updates a User if it has one. This is a very confusing when looking at the code as you must jump between two models, the oidc code and signals to understand what is happening when. Let's stop that! The only reason for continuing this was we want to acknoledge that users may get deleted, but per an eng huddle meeting we agreed users would get deactivated and thus soft deleted instead. In #2186 we made a script that will copy over all non-email values from a contact and copy it to it's appropriate user which fills the whole of making sure Users doesn't lose any of the data that was in Contacts. Now we can go ahead and stop using a User contact everywhere and change it to use just the user values. (Note we will delete the contacts that have user connections in a follow on ticket).

### Acceptance criteria

- [ ] Contacts are not deleted who have a user row (we need to keep them around)
- [ ] Change all references to Contact on a User to just use the User values. This especially includes in profile views.
- [ ] Remove the creation of contacts when a user is made (remove the signal) 
- [ ] remove the related signal documentation
- [ ] remove the logic where a contact updates a user.  (don't delete the user reference on the contact model, as we want to be sure to remove these contacts later)
- [ ] your contact in the domain management forms now alter the user values not user.contact
- [ ] There should be no longer any references to a contact on a User object, including in unit tests

### Additional context

_No response_

### Links to other issues

blocked by #2186

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #2247 

## Changes

- Changed all references to Contact in views, including user profile view, finish setup view, and domain your contact information view
- Removed the creation of contacts when a user is made (signal was removed)
- Remove all signal documentation
- Updated import_tables script to account for removal of signals
- Removed logic where contact updates a user
- Removed connection between user and contact completely
- Removed Contact section from User edit form in Django admin
- Removed User from Contact edit form in Django admin
- Removed Is User column from Contact table in Django admin
- Removed script which updates users with associated contact information (**NOTE** this script will need to be run on stable before this PR is merged into stable)

## Context for reviewers

This change is rather low level to the Contact and User model and their relationship to each other. There are a few noticeable changes which should be tested below.  And to make the model changes, there was a fair bit of refactoring of existing functionality which should be retested. Below outlines the known changes which should be tested to verify the PR:

In Django admin:
 The Contacts list table should no longer display a Is User column
 When editing a Contact, there is no reference to an associated User
 When editing a User, there is no longer a section at the bottom for the associated Contact
User profile form updates the User table (rather than the Contact table)
Finish user setup form updates the User table (rather than the Contact table)
When a User record is created, no Contact record is automatically created
When a Contact record is updated, no User record is automatically updated
When a domain request is updated from the home page, orphaned contacts are properly deleted
When a contact in the domain form is updated, the logic that updates the contact in place or creates a new contact has been updated to account for there no longer being a connection between user and contact
Export / import tables scripts have been updated (this may be too much to test, and was a very small change, so you may decide not to verify this part)

## Setup

This PR requires running migrations before testing

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Comment by abroddrick** [comment]
@zandercymatics RIGHT before this one gets released to stable be sure to run the copy contacts script. This one removes that script from being an option, so you can no longer run the script after the release is made. CC: @therealslimhsiehdy

**Comment by gabydisarli** [comment]
@dave-kennedy-ecs I'll be doing design review.

**Review by gabydisarli — APPROVED** [review]

File: src/registrar/forms/domain.py [hunk 1]
```diff
@@ -203,6 +203,63 @@ def clean(self):
 )
 
 
+class UserForm(forms.ModelForm):
+    """Form for updating contacts."""
```
[the remark this answers is not in the document]
Comment (dave-kennedy-ecs) [reply src/registrar/forms/domain.py:207]: good catch

File: src/registrar/tests/test_admin.py [hunk 2]
```diff
@@ -2092,11 +2084,11 @@ def test_contact_fields_have_detail_table(self):
 
         # == Check for the creator == #
 
-        # Check for the right title, email, and phone number in the response.
+        # Check for the right title and phone number in the response.
+        # Email will appear more than once
         expected_creator_fields = [
             # Field, expected value
             ("title", "Treat inspector"),
-            ("email", "meoward.jones@igorville.gov"),
```
[the remark this answers is not in the document]
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_admin.py:2099]: yes

File: src/registrar/tests/test_admin.py [hunk 3]
```diff
@@ -3133,13 +3121,12 @@ def test_contact_fields_have_detail_table(self):
...
         # == Check for the creator == #
 
-        # Check for the right title, email, and phone number in the response.
+        # Check for the right title and phone number in the response.
         # We only need to check for the end tag
         # (Otherwise this test will fail if we change classes, etc)
         expected_creator_fields = [
             # Field, expected value
             ("title", "Treat inspector"),
-            ("email", "meoward.jones@igorville.gov"),
```
[the remark this answers is not in the document]
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_admin.py:3142]: yes

File: src/registrar/tests/test_admin.py [hunk 4]
```diff
@@ -4049,7 +4036,7 @@ def test_readonly_when_restricted_staffuser(self):
 
             readonly_fields = self.admin.get_readonly_fields(request)
 
-            expected_fields = ["user", "email"]
+            expected_fields = []
```
[the remark this answers is not in the document]
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_admin.py:4039]: yes

**Pull request merged**